File: 1400_construct_k_palindrome_strings.java
class Solution {
    public boolean canConstruct(String x, int y) {
        if (y > x.length()) return false;
        int[] z = new int[26];
        int c = 0;
        for (char a : x.toCharArray()) z[a - 'a']++;
        for (int b : z) c += b % 2;
        return c <= y;
    }
}

File: 1400_construct_k_palindrome_strings.cpp
class Solution {
public:
    bool canConstruct(string z, int x) {
        if (x > z.size()) return false;
        int y[26] = {};
        for (char c : z) y[c - 'a']++;
        int w = 0;
        for (int i : y) w += i % 2;
        return w <= x;
    }
};
